SdlLogs/SdlLogs.py

```diff
@@ -0,0 +1,159 @@
+			enterRegex = enterRegex[enterRegex.find(" ")+1:]
+			enterRegex = enterRegex[:enterRegex.find(" ")]
+
+			print (enterRegex)
+
+			if enterRegex is not "":
+				exitRegion = self.view.find(enterRegex+threadExit, posFatal)
+				print (self.view.substr(exitRegion))
+
+				if exitRegion.size() is not 0:
+					posFatal = enterRegion.b
+				else:
+					self.view.show(enterRegion)
+					posFatal = enterRegion.b
+					break
+
+
+
+
+class HideThreadAdressCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		global actThreadFlag
+		trcRegions = self.view.find_all(threadRegex)
+
+		if actThreadFlag is False:
+			self.view.fold(trcRegions)
+			actThreadFlag = True
+		else:
+			self.view.unfold(trcRegions)
+			actThreadFlag = False
+
+
+class HideComponentCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		global actComponentFlag
+		trcRegions = self.view.find_all(componentRegex)
+
+		if actComponentFlag is False:
+			self.view.fold(trcRegions)
+			actComponentFlag = True
+		else:
+			self.view.unfold(trcRegions)
+			actComponentFlag = False
+
+
+class HideExtraPathCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		global actPathFlag
+		trcRegions = self.view.find_all(junkRegex)
+
+		if actPathFlag is False:
+			self.view.fold(trcRegions)
+			actPathFlag = True
+		else:
+			self.view.unfold(trcRegions)
+			actPathFlag = False
+
+
+class FilterByValueCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		selTextRegex = getSelectedText(self)
+		if selTextRegex is not "":
+			trcRegions = self.view.find_all(selTextRegex)
+			cutRegion = sublime.Region(0, 0)
+			leftBorder = 0
+
+			for i in range (0, len(trcRegions)):
+				trcRegions[i] = self.view.full_line(trcRegions[i])
+				trcRegions[i] = self.view.substr(trcRegions[i])
+
+			self.view.erase(edit, sublime.Region(0, self.view.size()))
+			leftBorder = 0
+
+			for i in range (0, len(trcRegions)):
+				leftBorder+= self.view.insert(edit, leftBorder, trcRegions[i])
+
+
+class JumpToFileCommand(sublime_plugin.TextCommand):
+	def run(self, edit):
+		FULL_PATH_FILE	= "(/.*)(/sdl_core/src/.{0,}\\.(cc|h|cpp|hpp)):(\\d{1,})"
+		CORRECTED_PATH  = "(/[^/ ]*)+/?"
+
+		primaryPath = re.search(FULL_PATH_FILE, self.view.substr(self.view.line(self.view.sel()[0])))
+		isItPath = re.search(CORRECTED_PATH, self.view.substr(self.view.line(self.view.sel()[0])))
+
+		if self.view.settings().get(sourcePath) != "": 
+			if primaryPath :
+				file = self.view.settings().get(sourcePath) + primaryPath.group(2)
+				if path.isfile(file) : 
+					self.view.window().open_file(file + ":" + primaryPath.group(4), sublime.ENCODED_POSITION)
+				else:
+					sublime.error_message("File '{0}' not found. Maybe path to source files in your settings is incorrect or file really absent".format(file))
+			elif isItPath:
+				sublime.error_message("Expected '/sdl_core/src/' in '{0}' ".format(isItPath.group(0)))
+		else:
+			if primaryPath :
+				file = primaryPath.group(1) + primaryPath.group(2)
```

Comment:
@dkandyba this code looks similar. May be extracted to a function?

Reply:
@LuxoftAKutsan fixed in c95c56eb0fcc3c822bc445ccb58a40238b9d2ff2